### DISCUSSION TITLE
collect serverStatus, Oplog and others metrics without --collect-all

### DISCUSSION THREAD

**Discussion by DBaffaleuf-Tac** [percona/mongodb_exporter#936] [Q&A] [answered]
Hi there,

I am testing an exporter in a docker container using --collect-all, but this is consuming way too much CPU on the target, so I am trying to find a way to limit the number of metrics collected by the exporter and enabling only the ones I need : 

```
docker run -d --name "percona-mg-exporter" -p 9216:9216 \
	--expose 9216 percona/mongodb_exporter:0.41.1 \ 
	--mongodb.uri="mongodb://mongodb-percona-mg-exporter:passwd@host:27017/admin?ssl=false" \
	--mongodb.collstats-colls=base.col1,base.col2 \ 
	--collector.pbm \
	--collector.dbstats
	--collector.collstats 
```

Yet there are couple of them I can't find in the documentation : https://github.com/percona/mongodb_exporter/blob/main/REFERENCE.md 

**1) Metrics related to db.serverStatus():**
_mongodb_ss_*_

**2) Metrics related to oplog stats:**
 _mongodb_oplog_stats_*_

**3) and this one :** _mongodb_members_configTerm_

Is there any ay to collect these counters without having to collect everything (especially the serverStatus ones) ? 
Thanks,

**Comment by BupycHuk** [comment, marked as answer]
Hi, can you try to enable diagnosticdata collector?

**Comment by DBaffaleuf-Tac** [comment]
Hi thanks it worked , I now have all the metrics I need :

```
docker run -d --name "percona-mg-exporter" -p 9216:9216 \ 
--expose 9216 percona/mongodb_exporter:0.41.1 \ 
--mongodb.uri="mongodb://mongodb-percona-mg-exporter:password@host:27017/admin?ssl=false" \
--mongodb.collstats-colls=base1.col1,base1.col2 \
--collector.pbm \
--collector.diagnosticdata \
--collector.dbstats \
--collector.collstats \
--collector.replicasetstatus
```

However, it seems to be leaking CPU% and MEM% over time. I restarted the container yesterday evening, consuming :

```
# ps -p $(pidof mongodb_exporter) -o %cpu,%mem,cmd
%CPU %MEM CMD
 0.7  0.2 /mongodb_exporter ...
```

And this morning :

```
ps -p $(pidof mongodb_exporter) -o %cpu,%mem,cmd
%CPU %MEM CMD
14.0  8.4 /mongodb_exporter ... 
```

**Reply by DBaffaleuf-Tac** [reply]
It looks like connections are leaked. I had 1400+ connections to the node at some point, I restarted the container, now I am back at 11 connections, could this be related to the GO Driver (1.16.1) ?

**Reply by BupycHuk** [reply]
We fixed connection leak in 0.41.2

**Reply by BupycHuk** [reply]
We found root cause of the problem and released v0.42.0.
